### DISCUSSION TITLE
Options line within PHP

### DISCUSSION THREAD

**Discussion by melTr** [add2cal/add-to-calendar-button#321] [Q&A] [answered]
Hi,
Thanks for your script, it works very well. However, I'm stuck on a small detail.
I'm trying to create a PHP function to integrate the button into my website. I'm struggling with the line 
`options="'Apple','Google','iCal','Outlook.com','Yahoo'"`

I tried many variations with ' or ", interchanging them. But either nothing appears (with PHP error about strings). If I put only one choice, the button work without problem; so the problem concerns ' or ".

The function looks like this (I removed some config to keep minimal info): 
```php
function BoutonCalendar (Page $item) {
  $out ='';
  $out .= '<add-to-calendar-button
  name="{$item->title}"
  options=" 'Apple','Google','iCal','Outlook.com','Yahoo' "
  ></add-to-calendar-button>';

return $out;
}
```

Does someone already implement this script in PHP?
Thanks
Mel

EDIT: sorry, finally found the solution. In case, it helps someone :   options='\"Apple\", \"Google\", \"Outlook.com\", \"Yahoo\", \"ical\"'

**Comment by jekuer** [comment, marked as answer]
Glad you solved it yourself.

Basically, you need to make sure that you do not use any single or double quotes around the element, which could interfere with it.
There are multiple ways of achieve this.

One strong solution in your case would look like this:

```php
function BoutonCalendar (Page $item) {
  $optionsDefault = "'Apple','Google','iCal','Outlook.com','Yahoo'";
  $out = '';
  $out .= '<add-to-calendar-button name="' . $item->title . '" options="' . $optionsDefault . '"';
  
  return $out;
}
```
